scripts/hub-cluster/kind/kind.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+KIND_VERSION="0.17.0"
+
+__dir=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+
+function check() {
+	if [ "$(kind --version)" == "kind version $KIND_VERSION" ]; then
+		return 0
+	else
+		echo "Does not have 'kind' with version $KIND_VERSION!"
+		return 1
+	fi
+}
+
+function install() {
+	if check; then
+		return 0
+	fi
+
+	echo "Installing kind $KIND_VERSION..."
+	curl -L https://kind.sigs.k8s.io/dl/v$KIND_VERSION/kind-linux-amd64 -o /usr/local/bin/kind
+	chmod u+x /usr/local/bin/kind
+	echo "Installed successfully!"
+}
+
+function setup_contour() {
+	# based on https://kind.sigs.k8s.io/docs/user/ingress/#contour
+	# for more information about contour, see: https://projectcontour.io
+	kubectl apply -f https://projectcontour.io/quickstart/contour.yaml
+	kubectl patch daemonsets -n projectcontour envoy -p \
+		'{"spec":{"template":{"spec":{"nodeSelector":{"ingress-ready":"true"},"tolerations":[{"key":"node-role.kubernetes.io/control-plane","operator":"Equal","effect":"NoSchedule"},{"key":"node-role.kubernetes.io/master","operator":"Equal","effect":"NoSchedule"}]}}}}'
+	timeout 2m bash -c 'until kubectl wait --for=condition=ready pods -n projectcontour --selector=app=contour; do sleep 5s; done'
```

Comment:
to check: maybe it would be more reliable to wait for the rollout for the deployment/daemonset as we did here:
https://github.com/openshift/assisted-service/pull/4489

Reply:
This is a great idea!
And yes, it's working quite good